src/context/products_context.js

```diff
@@ -13,18 +13,72 @@ import {
   GET_SINGLE_PRODUCT_ERROR,
 } from '../actions'
 
-const initialState = {}
+const initialState = {
+  isSidebarOpen: false,
+  products_loading: false,
+  products_error: false,
+  products: [],
+  featured_products: [],
+  single_product_loading: false,
+  single_product_error: false,
+  single_product: {},
+}
 
 const ProductsContext = React.createContext()
 
 export const ProductsProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+
+  const openSidebar = () => {
+    dispatch({ type: SIDEBAR_OPEN })
+  }
+  const closeSidebar = () => {
+    dispatch({ type: SIDEBAR_CLOSE })
+  }
+
+  const fetchProducts = async (url) => {
+    dispatch({ type: GET_PRODUCTS_BEGIN })
+    try {
+      const response = await axios.get(url)
+      const products = response.data.products
+      dispatch({ type: GET_PRODUCTS_SUCCESS, payload: products })
+    } catch (error) {
+      dispatch({ type: GET_PRODUCTS_ERROR })
+    }
+  }
+//fn that fetches single product- without line 55 I used to get an array of products just like in fetchProducts fn , now it gives an error in console bundle.js:146779     GET http://localhost:3000/products/NaN 404 (Not Found)- so in SingleProductPage.js i converted it to number - line 18
+  const fetchSingleProduct = async (url,id) => {
+    dispatch({ type: GET_SINGLE_PRODUCT_BEGIN })
+    try {
+      const response = await axios.get(url/id)
```

Comment:
Here we're taking the variable `url` and _dividing it_ by the variable `id`. But url and id are both strings, so this is resulting in `NaN` (not a number) as the result.  
 

Eg if you put above this line: `console.log("GET: ", url/id)` it would log in the devtools console, `GET: NaN`.

Then when you pass that `NaN` to `axios.get`,  it will try to do a GET request on NaN, and since no URL host was defined, axios will just use the current host (so localhost:3000 if you're running this locally)

  
Instead, what I think you intended to do was to interpolate those variables in a _string_. So something like:

```suggestion
      const response = await axios.get(`${url}/${id}`)
```

Reply:
@akosasante  I knew something was wrong with what I pass here, I saw that NaN in devtools, I also tried the string value but it didn't help because there are more things to change!!! thank you so much for explaining this!